### ISSUE TITLE
CF template FN::ImportValue fails on parameter validation.

### ISSUE THREAD

**Issue by ssanri** [localstack/localstack#3184]
Hi, it looks like the issue with exports is fixed but still referring exports from lambda issue persists. Still getting 

```
2020-10-27T03:41:28:WARNING:localstack.utils.cloudformation.template_deployer: Unexpected error retrieving details for resource ApiGateway::Resource: Parameter validation failed:

Invalid type for parameter restApiId, value: {'Fn::ImportValue': 'ApiGateway-local-restApiId'}, type: <class 'dict'>, valid types: <class 'str'> Traceback (most recent call last):
File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 819, in retrieve_resource_details
api_resources = aws_stack.connect_to_service('apigateway').get_resources(restApiId=api_id)['items']
File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 357, in _api_call
return self._make_api_call(operation_name, kwargs)
File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 648, in _make_api_call
request_dict = self._convert_to_request_dict(
File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 696, in _convert_to_request_dict
request_dict = self._serializer.serialize_to_request(
File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/validate.py", line 297, in serialize_to_request
raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter restApiId, value: {'Fn::ImportValue': 'ApiGateway-local-restApiId'}, type: <class 'dict'>, valid types: <class 'str'>

- {'Type': 'AWS::ApiGateway::Resource', 'Properties': {'ParentId': {'Fn::ImportValue': 'ApiGateway-local-2-4-restApiRootId'}, 'PathPart': 'healthcheck', 'RestApiId': {'Fn::ImportValue': 'ApiGateway-local-restApiId'}}} {'StackName': 'test-2-4-local', 'StackId': 'arn:aws:cloudformation:us-east-1:000000000000:stack/test-2-4-local/243177bf-a282-49c6-97e1-518fe2d3748e', 'LogicalResourceId': 'ApiGatewayResourceHealthcheck', 'PhysicalResourceId': '', 'ResourceType': 'AWS::ApiGateway::Resource', 'ResourceStatus': 'UPDATE_COMPLETE'}
```

error when I try to refer an exported output.

```
 aws --endpoint-url=http://localhost:4566 cloudformation list-exports
{
    "Exports": [
        {
            "ExportingStackId": "arn:aws:cloudformation:us-east-1:000000000000:stack/api-gateway-local/45b90c84-8539-4408-8f41-35b67c91652a",
            "Name": "ApiGateway-local-restApiId",
            "Value": "b68thqr1qw"
        },
        ...
]
```

_Originally posted by ssanri in https://github.com/localstack/localstack/issues/2041#issuecomment-716963016_

┆Issue is synchronized with this [Jira Task](https://localstack.atlassian.net/browse/LOC-76) by [Unito](https://www.unito.io/learn-more)

**Comment by ssanri** [comment]
oh wow,  days passed not even a comment from localstack maintainers. Am I the only one using Fn functions to get resource outputs from cloudformation? That's odd.

**Comment by duongpv7** [comment]
Thanks for reporting @ssanri. Can you provide the template `serverless.yml` to help us reproduce this issue?

**Comment by ssanri** [comment]
You can try with sample Serverless AWS Shared APIGateway example project. When deploying users or transactions lambdas, localstack is returning same error where api gateway parameter trying to get `SharedGW-restApiID` resource output

```
...
apiGateway: 
     restApiId: 
         "Fn::ImportValue": SharedGW-restApiId
     restApiRootResourceId: 
         "Fn::ImportValue": SharedGW-rootResourceId
...
```

**Comment by franklymrshankley** [comment]
I also have this issue, which manifests in exactly the same way.

**Comment by whummer** [comment]
Thanks for reporting @ssanri @franklymrshankley . This should be fixed in #3286 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by franklymrshankley** [comment]
Thanks for investigating this, much appreciated.

It feels like I'm a step closer now, in that my Lambdas actually get deployed and can be invoked manually via sls invoke -f <fnName>.

I still can't call them via a shared API gateway, and I think something is failing at the point at which the handlers are being registered with the gateway.

Apologies for the horrendous copy'n'paste, but this is the output I get from localstack during an `sls deploy` of the functions, after the API GW has successfully deployed.  There are no apparent errors returned from sls deploy, even with SLS_DEBUG=* turned on.  The stacks I'm attempting to deploy are from the example shared gateway, linked above.


`2020-11-30T21:01:00:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource ApiGatewayResourceUsersList cannot be deployed, found unsatisfied dependencies. ['ApiGatewayResourceUsers'] - {'Type': 'AWS::ApiGateway::Resource', 'Properties': {'ParentId': {'Ref': 'ApiGatewayResourceUsers'}, 'PathPart': 'list', 'RestApiId': {'Fn::ImportValue': 'SharedGW-restApiId'}}} (existing: {'ServerlessDeploymentBucket', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Updating resource ID from cda18386 to 4yzcqdjsqf (eu-west-2)
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource ApiGatewayDeployment1606770054020 cannot be deployed, found unsatisfied dependencies. ['ApiGatewayMethodUsersListGet'] - {'Type': 'AWS::ApiGateway::Deployment', 'Properties': {'RestApiId': {'Fn::ImportValue': 'SharedGW-restApiId'}, 'StageName': 'local'}, 'DependsOn': ['ApiGatewayMethodUsersListGet']} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource GetDashusersLambdaVersionAYbmSpywl7x0lX3cdPlFdNcuLTPWMN4OPY5cwzM3Ul4 cannot be deployed, found unsatisfied dependencies. ['GetDashusersLambdaFunction'] - {'Type': 'AWS::Lambda::Version', 'DeletionPolicy': 'Retain', 'Properties': {'FunctionName': {'Ref': 'GetDashusersLambdaFunction'}, 'CodeSha256': 'kmJkJZbJyL/UkHhffdPVKKG6QoVi1sV9BEEh9GCH2B4='}} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource GetDashusersLambdaFunction cannot be deployed, found unsatisfied dependencies. ['ServerlessDeploymentBucket'] - {'Type': 'AWS::Lambda::Function', 'Properties': {'Code': {'S3Bucket': {'Ref': 'ServerlessDeploymentBucket'}, 'S3Key': 'serverless/eshop-users/local/1606770055460-2020-11-30T21:00:55.460Z/eshop-users.zip'}, 'FunctionName': 'eshop-users-local-get-users', 'Handler': 'handler.getUsers', 'MemorySize': 1024, 'Role': {'Fn::GetAtt': ['IamRoleLambdaExecution', 'Arn']}, 'Runtime': 'nodejs12.x', 'Timeout': 6}, 'DependsOn': ['GetDashusersLogGroup']} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource GetDashusersLambdaPermissionApiGateway cannot be deployed, found unsatisfied dependencies. ['GetDashusersLambdaFunction'] - {'Type': 'AWS::Lambda::Permission', 'Properties': {'FunctionName': {'Fn::GetAtt': ['GetDashusersLambdaFunction', 'Arn']}, 'Action': 'lambda:InvokeFunction', 'Principal': 'apigateway.amazonaws.com', 'SourceArn': {'Fn::Join': ['', ['arn:', {'Ref': 'AWS::Partition'}, ':execute-api:', {'Ref': 'AWS::Region'}, ':', {'Ref': 'AWS::AccountId'}, ':', {'Fn::ImportValue': 'SharedGW-restApiId'}, '/*/*']]}}} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource ApiGatewayMethodUsersListGet cannot be deployed, found unsatisfied dependencies. ['GetDashusersLambdaFunction'] - {'Type': 'AWS::ApiGateway::Method', 'Properties': {'HttpMethod': 'GET', 'RequestParameters': {}, 'ResourceId': {'Ref': 'ApiGatewayResourceUsersList'}, 'RestApiId': {'Fn::ImportValue': 'SharedGW-restApiId'}, 'ApiKeyRequired': False, 'AuthorizationType': 'NONE', 'Integration': {'IntegrationHttpMethod': 'POST', 'Type': 'AWS_PROXY', 'Uri': {'Fn::Join': ['', ['arn:', {'Ref': 'AWS::Partition'}, ':apigateway:', {'Ref': 'AWS::Region'}, ':lambda:path/2015-03-31/functions/', {'Fn::GetAtt': ['GetDashusersLambdaFunction', 'Arn']}, '/invocations']]}}, 'MethodResponses': []}} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource GetDashusersLambdaVersionAYbmSpywl7x0lX3cdPlFdNcuLTPWMN4OPY5cwzM3Ul4 cannot be deployed, found unsatisfied dependencies. ['GetDashusersLambdaFunction'] - {'Type': 'AWS::Lambda::Version', 'DeletionPolicy': 'Retain', 'Properties': {'FunctionName': {'Ref': 'GetDashusersLambdaFunction'}, 'CodeSha256': 'kmJkJZbJyL/UkHhffdPVKKG6QoVi1sV9BEEh9GCH2B4='}} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsers', 'IamRoleLambdaExecution', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Updating resource ID from b6defcff to yoilj5ykpf (eu-west-2)
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource ApiGatewayMethodUsersListGet cannot be deployed, found unsatisfied dependencies. ['GetDashusersLambdaFunction'] - {'Type': 'AWS::ApiGateway::Method', 'Properties': {'HttpMethod': 'GET', 'RequestParameters': {}, 'ResourceId': {'Ref': 'ApiGatewayResourceUsersList'}, 'RestApiId': {'Fn::ImportValue': 'SharedGW-restApiId'}, 'ApiKeyRequired': False, 'AuthorizationType': 'NONE', 'Integration': {'IntegrationHttpMethod': 'POST', 'Type': 'AWS_PROXY', 'Uri': {'Fn::Join': ['', ['arn:', {'Ref': 'AWS::Partition'}, ':apigateway:', {'Ref': 'AWS::Region'}, ':lambda:path/2015-03-31/functions/', {'Fn::GetAtt': ['GetDashusersLambdaFunction', 'Arn']}, '/invocations']]}}, 'MethodResponses': []}} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsersList', 'ApiGatewayResourceUsers', 'IamRoleLambdaExecution', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.cloudformation.cloudformation_starter: Resource ApiGatewayDeployment1606770054020 cannot be deployed, found unsatisfied dependencies. ['ApiGatewayMethodUsersListGet'] - {'Type': 'AWS::ApiGateway::Deployment', 'Properties': {'RestApiId': {'Fn::ImportValue': 'SharedGW-restApiId'}, 'StageName': 'local'}, 'DependsOn': ['ApiGatewayMethodUsersListGet']} (existing: {'ServerlessDeploymentBucket', 'ApiGatewayResourceUsersList', 'ApiGatewayResourceUsers', 'IamRoleLambdaExecution', 'AWS::Region', 'AWS::StackName', 'AWS::URLSuffix', 'ServerlessDeploymentBucketPolicy', 'AWS::Partition', 'GetDashusersLogGroup', 'AWS::StackId', 'AWS::AccountId'})
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:INFO:localstack.services.awslambda.lambda_api: Function not found: arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users
2020-11-30T21:01:01:WARNING:localstack.utils.cloudformation.template_deployer: Unexpected resource type Lambda::Permission when resolving references of resource GetDashusersLambdaPermissionApiGateway: {'Type': 'AWS::Lambda::Permission', 'Properties': {'FunctionName': {'Fn::GetAtt': ['GetDashusersLambdaFunction', 'Arn']}, 'Action': 'lambda:InvokeFunction', 'Principal': 'apigateway.amazonaws.com', 'SourceArn': {'Fn::Join': ['', ['arn:', {'Ref': 'AWS::Partition'}, ':execute-api:', {'Ref': 'AWS::Region'}, ':', {'Ref': 'AWS::AccountId'}, ':', {'Fn::ImportValue': 'SharedGW-restApiId'}, '/*/*']]}}}
2020-11-30T21:01:01:WARNING:localstack.utils.cloudformation.template_deployer: Unexpected resource type Lambda::Permission when resolving references of resource GetDashusersLambdaPermissionApiGateway: {'Type': 'AWS::Lambda::Permission', 'Properties': {'FunctionName': 'arn:aws:lambda:eu-west-2:000000000000:function:eshop-users-local-get-users', 'Action': 'lambda:InvokeFunction', 'Principal': 'apigateway.amazonaws.com', 'SourceArn': 'arn:aws:execute-api:eu-west-2:000000000000:yn9ypab3sd/*/*'}}
2020-11-30T21:01:02:WARNING:localstack.services.cloudformation.cloudformation_starter: Unable to determine physical_resource_id for resource <class 'localstack_ext.services.cloudformation.service_models.GenericBaseObject'>
2020-11-30T21:01:02:WARNING:localstack.services.cloudformation.cloudformation_starter: Unable to determine physical_resource_id for resource <class 'moto.apigateway.models.Method'>
2020-11-30T21:01:02:INFO:localstack.services.cloudformation.cloudformation_starter: Updating resource ID from b839e292 to t8aflafmwt (eu-west-2)
2020-11-30T21:01:02:WARNING:localstack.services.cloudformation.cloudformation_starter: Unable to determine physical_resource_id for resource <class 'moto.apigateway.models.Deployment'>
`
